wicketd/src/context.rs

```diff
@@ -7,11 +7,45 @@
 use crate::artifacts::WicketdArtifactStore;
 use crate::update_tracker::UpdateTracker;
 use crate::MgsHandle;
+use gateway_client::types::SpIdentifier;
+use gateway_client::types::SpState;
+use gateway_client::types::SpType;
+use sled_hardware::Baseboard;
 
 /// Shared state used by API handlers
 pub struct ServerContext {
     pub mgs_handle: MgsHandle,
     pub mgs_client: gateway_client::Client,
     pub(crate) artifact_store: WicketdArtifactStore,
     pub(crate) update_tracker: UpdateTracker,
+    pub(crate) baseboard: Baseboard,
+}
+
+impl ServerContext {
+    pub(crate) fn can_update_target_sp(
+        &self,
+        id: SpIdentifier,
+        state: &SpState,
+    ) -> bool {
+        // If we failed don't know our own baseboard, fall back to just using
+        // the id, from which we know whether or not the target sled is one
+        // of the scrimlets. If it is, refuse to update, because it might be
+        // where we're running now.
+        if self.baseboard == Baseboard::unknown() {
```

Comment:
This state does seem kinda like a pain to deal with -- could we just make the baseboard information non-optional, and have wicketd refuse to start if the correct baseboard info is not identified?

Reply:
It would be very easy from wicketd's point of view to remove this check and just assume sled-agent gave it something valid. Do you know what you expect sled-agent to do if/when you remove `Baseboard::unknown()` and it has problems reading the baseboard info?

Reply:
To clarify, I think that it's the responsibility of the SMF `start` method + the CLI argument parsing to catch the case where `unknown` was provided, and opt to throw an error.

My only suggestion here is just to bail out *earlier* in the process, so that the `wicketd` implementation doesn't need to deal with something that is functionally `Option<Baseboard>`.

Reply:
> Hmm, that seems a little weird - either wicket's CLI argument parsing now has special checks for the `Baseboard::unknown()` values that are themselves placeholders

This is kinda what I was thinking - we're doing the same thing here, but it's  further away from the argument parsing.

Reply:
Ok, I reworked this in [bc7aed0](https://github.com/oxidecomputer/omicron/pull/2940/commits/bc7aed05b88f5cba709d97200da0a3d93995b97e). There's still one call to `Baseboard::unknown()`, but it's in wicketd's main and has a TODO that it will need to be removed at some point. The rest of wicketd traffics solely in `Option<Baseboard>`.